src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -0,0 +1,210 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.direction.EdgeDirectionComparator;
+
+/**
+ * This check identifies Edges with no name Tag that are between two other Edges with the same name
+ * Tag, OR the Edge has a name Tag but does not equal the name Tag of the Edges that it is between.
+ *
+ * @author sugandhimaheshwaram
+ */
+public class RoadNameGapCheck extends BaseCheck<Long>
+{
+    /**
+     * This checks if the connected edges have same direction or not.
+     */
+    static class EdgePredicate
+    {
+        private Edge edge;
+        private EdgeDirectionComparator edgeDirectionComparator = new EdgeDirectionComparator();
+
+        EdgePredicate(final Edge edge)
+        {
+            this.edge = edge;
+        }
+
+        boolean isSameHeading(final AtlasObject object)
+        {
+            final Edge otherEdge = (Edge) object;
+            return object != null
+                    && this.edgeDirectionComparator.isSameDirection(this.edge, otherEdge, false);
+        }
+    }
+
+    // You can use serialver to regenerate the serial UID.
+    private static final long serialVersionUID = 7104778218412127847L;
+    private static final List<String> VALID_HIGHWAY_TAG_DEFAULT = Arrays.asList("primary",
+            "secondary", "tertiary", "trunk", "motorway");
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList("Edge name is empty.",
```

Comment:
nit: It would be useful to have the flagged object's identifier in the instruction.